trace/trace.go

```diff
@@ -28,29 +28,43 @@ import (
 type traceStep struct {
 	stepTime time.Time
 	msg      string
+	msgArgs  []interface{}
 }
 
 // Trace keeps track of a set of "steps" and allows us to log a specific
 // step if it took longer than its share of the total allowed time
 type Trace struct {
 	name      string
+	msg       string
```

Comment:
Why do we need `msg` and `msgArgs` in addition to `name`? Can we just have `name` + `nameArgs` instead?

Reply:
That's one option. Looking at the use cases, having two fields: "name" and formatted "details" seems like it might be a good fit. I've updated the PR. Let me know what you think.

Reply:
sounds good to me